RationaleMCP/0027/HindleyMilner.md

```diff
@@ -0,0 +1,336 @@
+# Unit checking
+
+Unit checking is the process of inferring the units of Modelica variables, and ensuring the variables are used consistently in expressions and equations.
+
+It takes place after any evaluation of evaluable parameters and no evaluable parameters shall be evaluated during this process.
```

Comment:
Clarify - it is important:
- ThermoFluidStream has units depending on evaluable parameters (different if-branches with evaluable condition)
- Exponents

Alternative implementation: - collect unit-constraints from original equations based on evaluated parameters; should give same result.
@qlambert-pro has variant text for this